apps/rxjs/49-hold-to-save-button/src/app/app.component.ts

```diff
@@ -1,25 +1,32 @@
-import { ChangeDetectionStrategy, Component } from '@angular/core';
+import { ChangeDetectionStrategy, Component, signal } from '@angular/core';
+import { HoldableDirective } from './holdable.directive';
 
 @Component({
   standalone: true,
-  imports: [],
+  imports: [HoldableDirective],
   selector: 'app-root',
   template: `
     <main class="flex h-screen items-center justify-center">
       <div
         class="flex w-full max-w-screen-sm flex-col items-center gap-y-8 p-4">
         <button
+          [appHoldable]="TRIGGER_TIME"
+          (appHoldableTime)="elapsed.set($event)"
+          (appHoldableDone)="onSend()"
           class="rounded bg-indigo-600 px-4 py-2 font-bold text-white transition-colors ease-in-out hover:bg-indigo-700">
           Hold me
         </button>
 
-        <progress [value]="20" [max]="100"></progress>
+        <progress [value]="elapsed()" [max]="TRIGGER_TIME"></progress>
```

Comment:
nit: you can use self-closing tag.

Reply:
True. I missed that.

I should add the eslint rule :)